softlayer/resource_softlayer_global_ip.go

```diff
@@ -0,0 +1,279 @@
+	if err != nil {
+		// Find price items with AdditionalServices
+		productOrderContainer, err = buildGlobalIpProductOrderContainer(d, sess, AdditionalServicesPackageType)
+		if err != nil {
+			return fmt.Errorf("Error creating global ip: %s", err)
+		}
+	}
+
+	log.Println("[INFO] Creating global ip")
+
+	receipt, err := services.GetProductOrderService(sess).
+		PlaceOrder(productOrderContainer, sl.Bool(false))
+	if err != nil {
+		return fmt.Errorf("Error during creation of global ip: %s", err)
+	}
+
+	globalIp, err := findGlobalIpByOrderId(sess, *receipt.OrderId)
+	d.SetId(fmt.Sprintf("%d", *globalIp.Id))
+	return resourceSoftLayerGlobalIpUpdate(d, meta)
+
+}
+
+func resourceSoftLayerGlobalIpRead(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(*session.Session)
+	service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+	globalIpId, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+	}
+
+	globalIp, err := service.Id(globalIpId).Mask(GlobalIpMask).GetObject()
+	if err != nil {
+		if apiErr, ok := err.(sl.Error); ok && apiErr.StatusCode == 404 {
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("Error retrieving Global Ip: %s", err)
+	}
+
+	d.Set("id", *globalIp.Id)
+	d.Set("ip_address", *globalIp.IpAddress.IpAddress)
+	d.Set("routes_to", *globalIp.DestinationIpAddress.IpAddress)
+
+	return nil
+}
+
+func resourceSoftLayerGlobalIpUpdate(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(*session.Session)
+	service := services.GetNetworkSubnetIpAddressGlobalService(sess)
+
+	globalIpId, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("Not a valid global ip ID, must be an integer: %s", err)
+	}
+
+	if d.HasChange("routes_to") {
+		_, err = service.Id(globalIpId).Route(sl.String(d.Get("routes_to").(string)))
+		if err != nil {
+			return fmt.Errorf("Error editing Global Ip: %s", err)
+		}
+		stateConf := &resource.StateChangeConf{
+			Pending: []string{"pending"},
+			Target:  []string{"complete"},
+			Refresh: func() (interface{}, string, error) {
+				subnetIpAddress, err := service.Id(globalIpId).GetDestinationIpAddress()
+				if err != nil {
+					return datatypes.Network_Subnet_IpAddress_Global{}, "pending", err
+				}
+
+				if *subnetIpAddress.IpAddress == d.Get("routes_to").(string) {
+					return datatypes.Network_Subnet_IpAddress_Global{}, "complete", nil
+				} else {
+					return nil, "pending", nil
+				}
+			},
+			Timeout:    10 * time.Minute,
+			Delay:      5 * time.Second,
+			MinTimeout: 3 * time.Second,
+		}
+
+		pendingResult, err := stateConf.WaitForState()
+
+		if err != nil {
+			return fmt.Errorf("Error waiting for global ip destination ip address to become active: %s", err)
+		}
+
+		if _, ok := pendingResult.(datatypes.Network_Subnet_IpAddress_Global); ok {
+			return nil
+		}
+	}
+
+	return resourceSoftLayerGlobalIpRead(d, meta)
```

Comment:
Don't think we should call _Read_ from _Update_. I know that it is to compute the fields upon Creation, since _Update_ is called from _Create_ due to how *routes_to* gets set after the fact. Instead, I think that  `findGlobalIpByOrderId` should return a GlobalIP that already has the _ip address_ set. That way we can compute (`d.Set()`) it from _Create_, and thus _Update_ doesn't need to call _Read_ anymore.

Otherwise _Read_ would always be called on subsequent _Update_s unnecessary.